podpac/core/compositor/compositor.py

```diff
@@ -215,6 +215,27 @@ def iteroutputs(self, coordinates, _selector=None):
             for src in sources:
                 yield src.eval(coordinates, _selector=_selector)
 
+    @common_doc(COMMON_COMPOSITOR_DOC)
+    def eval(self, coordinates, **kwargs):
+        super_coordinates = coordinates
+
+        # remove extra dimensions
+        if self.dims:
+            extra = [
+                c.name
+                for c in coordinates.values()
+                if (isinstance(c, Coordinates1d) and c.name not in self.dims)
+                or (isinstance(c, StackedCoordinates) and all(dim not in self.dims for dim in c.dims))
+            ]
+            super_coordinates = super_coordinates.drop(extra)
+
+        output = super().eval(super_coordinates, **kwargs)
```

Comment:
THAT is fantastic! Get infront of the eval that actually does the caching! Brilliant!

Reply:
Actually, that's tricky enough that you should add a comment explaining why you suddenly have an `eval` function showing up instead of the usual `_eval` function.